source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -115,13 +158,80 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   return Http::FilterHeadersStatus::StopIteration;
 }
 
+Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers, bool) {
+  const auto* config = getConfig();
+
+  if (config->enableXRateLimitHeaders()) {
```

Comment:
Do we want to also check `if (!config->enabled())` here?

Also, can we avoid `absl::optional vector<LocalDescriptor>` and `push(absl::nullopt)` by having the check above?

Reply:
Yes, and it is changed.